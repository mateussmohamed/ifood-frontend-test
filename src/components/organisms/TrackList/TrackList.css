.track-list-container {
  padding: 3rem;
}

.track-list-title {
  color: #7380af;
  font-size: 16px;
  font-weight: normal;
}

.track-list {
  padding: 0;
}
