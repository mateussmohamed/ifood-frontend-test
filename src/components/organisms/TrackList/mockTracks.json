{
  "href":
    "https://api.spotify.com/v1/users/spotify/playlists/37i9dQZF1DX34KNiFQJNVm/tracks?offset=1&limit=1",
  "items": [
    {
      "added_at": "2017-12-01T02:51:17Z",
      "added_by": null,
      "is_local": false,
      "track": {
        "album": {
          "album_type": "single",
          "artists": [
            {
              "external_urls": {
                "spotify":
                  "https://open.spotify.com/artist/1mXAhKnZEdF6rotyyd4GBi"
              },
              "href":
                "https://api.spotify.com/v1/artists/1mXAhKnZEdF6rotyyd4GBi",
              "id": "1mXAhKnZEdF6rotyyd4GBi",
              "name": "MC Kevinho",
              "type": "artist",
              "uri": "spotify:artist:1mXAhKnZEdF6rotyyd4GBi"
            }
          ],
          "external_urls": {
            "spotify": "https://open.spotify.com/album/5Uy6cnLbM2D3x5v2CMWXM3"
          },
          "href": "https://api.spotify.com/v1/albums/5Uy6cnLbM2D3x5v2CMWXM3",
          "id": "5Uy6cnLbM2D3x5v2CMWXM3",
          "images": [
            {
              "height": 640,
              "url":
                "https://i.scdn.co/image/9b17af01c5bfe010c2bd11769f0dec1ae819446a",
              "width": 640
            },
            {
              "height": 300,
              "url":
                "https://i.scdn.co/image/36e59e5b7777958ac1594a3223f76142b2f573f9",
              "width": 300
            },
            {
              "height": 64,
              "url":
                "https://i.scdn.co/image/d3e03972d1c32a071bf1453a7fb063124fc40ca0",
              "width": 64
            }
          ],
          "name": "Rabiola",
          "type": "album",
          "uri": "spotify:album:5Uy6cnLbM2D3x5v2CMWXM3"
        },
        "artists": [
          {
            "external_urls": {
              "spotify":
                "https://open.spotify.com/artist/1mXAhKnZEdF6rotyyd4GBi"
            },
            "href": "https://api.spotify.com/v1/artists/1mXAhKnZEdF6rotyyd4GBi",
            "id": "1mXAhKnZEdF6rotyyd4GBi",
            "name": "MC Kevinho",
            "type": "artist",
            "uri": "spotify:artist:1mXAhKnZEdF6rotyyd4GBi"
          }
        ],
        "disc_number": 1,
        "duration_ms": 175660,
        "explicit": false,
        "external_ids": {
          "isrc": "BRKO11700182"
        },
        "external_urls": {
          "spotify": "https://open.spotify.com/track/228vRHzGfQh47EfLjuY1Yr"
        },
        "href": "https://api.spotify.com/v1/tracks/228vRHzGfQh47EfLjuY1Yr",
        "id": "228vRHzGfQh47EfLjuY1Yr",
        "name": "Rabiola",
        "popularity": 56,
        "preview_url":
          "https://p.scdn.co/mp3-preview/05f67a3f7433c2c638321ab98a0f5bcd1d14b7f1?cid=774b29d4f13844c495f206cafdad9c86",
        "track_number": 1,
        "type": "track",
        "uri": "spotify:track:228vRHzGfQh47EfLjuY1Yr"
      }
    }
  ],
  "limit": 1,
  "next":
    "https://api.spotify.com/v1/users/spotify/playlists/37i9dQZF1DX34KNiFQJNVm/tracks?offset=2&limit=1",
  "offset": 1,
  "previous":
    "https://api.spotify.com/v1/users/spotify/playlists/37i9dQZF1DX34KNiFQJNVm/tracks?offset=0&limit=1",
  "total": 50
}
