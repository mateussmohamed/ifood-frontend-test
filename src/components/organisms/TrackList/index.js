import React from 'react';
import PropTypes from 'prop-types';
import TrackRow from '../../molecules/TrackRow';

import './TrackList.css';

const convertToHumanTime = duration => {
  let s = parseInt((duration / 1000) % 60, 10);
  const m = parseInt((duration / (1000 * 60)) % 60, 10);
  s = s < 10 ? `0${s}` : s;
  return `${m}:${s}`;
};

const getLabels = track => {
  return {
    title: track.name,
    artist: track.artists.map(art => art.name).join(','),
    album: track.album.name,
    duration: convertToHumanTime(track.duration_ms)
  };
};

const TrackList = ({ data }) => (
  <section className="track-list-container">
    <h2 className="track-list-title">
      Daily playlist (created for you, by us)
    </h2>
    <ul className="track-list">
      {data.items.map(item => {
        const labels = getLabels(item.track);
        return <TrackRow labels={labels} key={`track-${item.track.name}`} />;
      })}
    </ul>
  </section>
);

TrackList.defaultProps = {
  data: {}
};

TrackList.propTypes = {
  data: PropTypes.objectOf(PropTypes.any)
};

export default TrackList;
