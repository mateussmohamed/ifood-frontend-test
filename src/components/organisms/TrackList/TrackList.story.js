import { storiesOf } from '@storybook/react';
import React from 'react';

import TrackList from './index';
import mockTracks from './mockTracks.json';

const stories = storiesOf('TrackList', module);

stories.add('default', () => (
  <section>
    <TrackList data={mockTracks} />
  </section>
));
