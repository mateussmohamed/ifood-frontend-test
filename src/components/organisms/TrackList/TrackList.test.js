import React from 'react';
import renderer from 'react-test-renderer';

import TrackList from './index';
import mockTracks from './mockTracks.json';

it('Should TrackList match to snapshot', () => {
  const component = <TrackList data={mockTracks} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
