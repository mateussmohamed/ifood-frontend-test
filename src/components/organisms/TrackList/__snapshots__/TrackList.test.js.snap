// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should TrackList match to snapshot 1`] = `
<section
  className="track-list-container"
>
  <ul
    className="track-list"
  >
    <li
      className="track-row"
    >
      <span
        className="track-label title"
      >
        Rabiola
      </span>
      <span
        className="track-label artist"
      >
        MC Kevinho
      </span>
      <span
        className="track-label album"
      >
        Rabiola
      </span>
      <span
        className="track-label duration"
      >
        2:55
      </span>
    </li>
  </ul>
</section>
`;
